openlibrary/plugins/upstream/edits.py

```diff
@@ -122,14 +58,131 @@ def GET(self):
             merge_requests=merge_requests,
         )
 
-    def extract_olids(self, url):
-        query_string = url.split('?')[1]
-        split_params = query_string.split('&')
-        params = {}
-        for p in split_params:
-            kv = p.split('=')
-            params[kv[0]] = kv[1]
-        return params['records'].split(',')
+    def POST(self):
+        data = json.loads(web.data())
+        rtype = data.get('rtype', '')
+        if rtype:
+            del data['rtype']
+
+        resp = process_merge_request(rtype, data)
+
+        return delegate.RawText(json.dumps(resp), content_type='application/json')
+
+    @staticmethod
+    def create_request(username, action='', type=None, olids='', comment: str = None):
```

Comment:
OPTIONAL: [`type()` is a Python builtin function](https://docs.python.org/3/library/functions.html#type) so shadowing it with a variable can be problematic.  A variable name such as `type_` avoids this.

Reply:
Thanks @cclauss!  Will change this, and the new `community_edits_queue` column, to `mr_type`.